src/replica/replica_2pc.cpp

```diff
@@ -77,6 +77,19 @@ void replica::on_client_write(dsn::message_ex *request, bool ignore_throttling)
         return;
     }
 
+    // validate partition_hash
```

Comment:
`reject_write_if_splitting`. If you want to eliminate the repeated code, you can use macro too:

```cpp
#define REJECT_REQUEST_IF_SPLITTING(op_type) \
if (_validate_partition_hash) { \
        if (_split_mgr->should_reject_request()) { \
            response_client##op_type##(request, ERR_SPLITTING); \
            return;
        }
        if (!_split_mgr->check_partition_hash( \
                ((dsn::message_ex *)request)->header->client.partition_hash, #op_type)) {
            response_client_write(request, ERR_PARENT_PARTITION_MISUSED);
            return;
        }
}  
```

Reply:
+1.

Reply:
As I comment to liwei, I think it clear to leave return error in `on_client_read` and `on_client_write`, and the repeated code is not too long.